### ISSUE TITLE
Support Next.js 14

### ISSUE THREAD

**Issue by binarybehemoth** [aws-amplify/amplify-hosting#3779]
### Is this feature request related to a new or existing Amplify category?

hosting

### Is this related to another service?

Next.js

### Describe the feature you'd like to request

Next.js 14 has been released.
Please make sure Amplify supports it, along with NodeJS 18+.
At the moment I cannot build it, after the system clones it from CodeCommit.

### Describe the solution you'd like

Smooth direct deployment.

### Describe alternatives you've considered

I am just deploying my Next.js app on my local PC at the moment.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change

**Comment by tarasowski** [comment]
+1 moved from Amplify to Vercel, Next 14 is not working right now.
Needs nodejs v18.

**Comment by shawnseaton** [comment]
+1 I can get Node.js 18 building with Next.js 14 by setting the build settings specifically to the Node 18.13.0 build image but it has symbolic link issues on the hosting platform :)

![image](https://github.com/aws-amplify/amplify-hosting/assets/22386863/84beed7b-b1ba-43f8-91d3-f3a533096442)

**Comment by siegerts** [comment]
Hi All, thanks for raising. Please see my comment here - https://github.com/aws-amplify/amplify-hosting/issues/3773#issuecomment-1792971715 .

Closing this as a duplicate to consolidate the conversation over there.

**Comment by kachkaev** [comment]
Next.js 14 requires at least Node 18.17 (see the list of breaking changes in [release notes](https://nextjs.org/blog/next-14#other-changes)).

I can't seem to get a working pipeline with any of these settings:


<img src="https://github.com/aws-amplify/amplify-hosting/assets/608862/35895513-c1d8-4813-9bfa-e8c375c07375" width=350>

```
2023-11-03T17:35:11.740Z [INFO]: # Retrieving environment cache...
2023-11-03T17:35:11.781Z [INFO]: # Retrieved environment cache
2023-11-03T17:35:11.781Z [INFO]: ---- Setting Up SSM Secrets ----
2023-11-03T17:35:11.782Z [INFO]: SSM params {"Path":"/amplify/d3bq67l0547585/dev/","WithDecryption":true}
2023-11-03T17:35:12.138Z [INFO]: # Determining if there are buildable backend changes in the amplify directory of the repository...
2023-11-03T17:35:12.150Z [INFO]: # Last built commit: 974d7afd4bb40f9946e9705a3e37ed6b6ffb2a54
2023-11-03T17:35:12.159Z [INFO]: # Differences detected.
2023-11-03T17:35:12.160Z [INFO]: # Continuing with backend build...
2023-11-03T17:35:12.167Z [WARNING]: Command failed: source ~/.bashrc 2> /dev/null || true && nvm current
                                    /bin/sh: 1: nvm: not found
2023-11-03T17:35:12.168Z [INFO]: # Patching NPM package '@aws-amplify/cli' to 12.7.1...
2023-11-03T17:35:12.453Z [INFO]: npm
2023-11-03T17:35:12.453Z [INFO]: ERR! `user` is not a valid npm option
2023-11-03T17:35:12.454Z [INFO]: 
2023-11-03T17:35:12.455Z [INFO]: npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-11-03T17_35_12_381Z-debug-0.log
2023-11-03T17:35:12.459Z [ERROR]: !!! Error patching NPM package '@aws-amplify/cli' to version 12.7.1. Verify the package name and version are correct, and retry the build. For help see our docs: https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html
2023-11-03T17:35:12.459Z [WARNING]: ! Unable to patch packages...
2023-11-03T17:35:12.460Z [INFO]: # Starting environment caching...
2023-11-03T17:35:12.460Z [INFO]: # Uploading environment cache artifact...
2023-11-03T17:35:12.553Z [INFO]: # Uploaded environment cache artifact
2023-11-03T17:35:12.553Z [INFO]: # Environment caching completed
```

---


<img src="https://github.com/aws-amplify/amplify-hosting/assets/608862/46d9b0b9-dcd6-410b-a6af-56b7aae069f8" width=350 >

```
2023-11-03T18:29:29.061Z [INFO]: # Retrieving environment cache...
2023-11-03T18:29:29.116Z [INFO]: # Retrieved environment cache
2023-11-03T18:29:29.116Z [INFO]: ---- Setting Up SSM Secrets ----
2023-11-03T18:29:29.116Z [INFO]: SSM params {"Path":"/amplify/d3bq67l0547585/dev/","WithDecryption":true}
2023-11-03T18:29:29.465Z [INFO]: # Determining if there are buildable backend changes in the amplify directory of the repository...
2023-11-03T18:29:29.477Z [INFO]: # Last built commit: 974d7afd4bb40f9946e9705a3e37ed6b6ffb2a54
2023-11-03T18:29:29.486Z [INFO]: # Differences detected.
2023-11-03T18:29:29.486Z [INFO]: # Continuing with backend build...
2023-11-03T18:29:29.494Z [ERROR]: !!! Node version not available: 18.18.2
2023-11-03T18:29:29.580Z [WARNING]: ! Unable to patch packages...
2023-11-03T18:29:29.581Z [INFO]: # Starting environment caching...
2023-11-03T18:29:29.581Z [INFO]: # Uploading environment cache artifact...
2023-11-03T18:29:29.642Z [INFO]: # Uploaded environment cache artifact
2023-11-03T18:29:29.642Z [INFO]: # Environment caching completed
Terminating logging...
```

---

18.13.0 seems to be the maximum available version on Amplify, with is too small for Next.js 14:
https://github.com/aws-amplify/amplify-hosting/issues/3109#issuecomment-1509264093
